### PR TITLE
new URL(userInput, requiredBase) is not good

### PR DESCRIPTION
**Pull request by bcaller**
It is possible for there to be a handful of false positives when string concatenation or interpolation is used for the `url` parameter but I don't think that will affect us much. If necessary, a more involved pattern can probably be written.

### CODE REVIEW

File: t3sts/semgrep_rules/url-constructor-base.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+// ok: url-constructor-base
+var notUnsafe3 = new URL(`/const`, location.origin);
+// todook: url-constructor-base
+var notUnsafe4 = new URL("?" + variable, constantOrVariable);
+// todook: url-constructor-base
+var notUnsafe5 = new URL(`/a${variable}/xyz`, "https://brave.com");
+// todook: url-constructor-base
+var notUnsafe6 = new URL("https://not.sure/" + variable, "https://brave.com");
+// todook: url-constructor-base
+var notUnsafe7 = new URL(`#${variable}`, constantOrVariable);
```
[the remark this answers is not in the document]
Comment (bcaller) [reply t3sts/semgrep_rules/url-constructor-base.js:39]: awkward

**Comment by thypon** [comment]
We should probably test for \\\\ and \\\/ and /\ which are equivalent to // in most parsers

**Comment by bcaller** [comment]
> We should probably test for \\ and \/ and /\ which are equivalent to // in most parsers

I added that to the message. Doesn't change the semgrep rule at all because the rule doesn't process the actual string.

**Review by thypon — APPROVED** [review]

**Comment by thypon** [comment]
We should fix the 502 error though :S

**Comment by bcaller** [comment]
Looks like I "fixed" it magically...

**Pull request merged**